src/sagas/contract.js

```diff
@@ -97,7 +98,40 @@ function* fetchContract({ payload: { contractAddress } }) {
     contract.disputeId
   )
 
-  yield put(contractActions.receiveContract({ ...disputeData, ...contract }))
+  let rulingData = null
+
+  if (contract.status === 4) {
+    try {
+      rulingData = yield call(
+        kleros.arbitrator.currentRulingForDispute,
+        contract.disputeId,
+        disputeData.numberOfAppeals
+      )
+    } catch (err) {
+      console.log(err)
+      throw new Error('Error current ruling failed')
+    }
+  }
+
+  let currentSession
+
+  try {
```

Comment:
Why not do the calls in parallel and try {} catch {} them all in the same block. So like 

```[rulingData, currentSession] = yield all([call(kleros.arbitrator.currentRulingForDispute, ...), call(kleros.arbitrator.getSession)])```